files/chefdk-env-customization/windows/chefdk_env_customization.rb

```diff
@@ -0,0 +1,52 @@
+## Environment hacks for running Ruby with ChefDK ##
+# ENV['HOME'] is not set by default on Windows. We need to set this to
+# something sensible since a lot of Ruby code depends on it. It is important 
+# for this directory to exist and be available, so we are introducing logic
+# here to pick a working HOME
+#
+# You can find this file in the repo at https://github.com/chef/omnibus-chef
+
+if !ENV['HOME'] || !File.exists?(ENV['HOME'])
+  old_home = ENV['HOME']
+  found = false
+  alternate_homes = []
+  alternate_homes << "#{ENV['HOMEDRIVE']}#{ENV['HOMEPATH']}" if ENV['HOMEDRIVE']
+  alternate_homes << "#{ENV['HOMESHARE']}" if ENV['HOMESHARE']
+  alternate_homes << "#{ENV['USERPROFILE']}" if ENV['USERPROFILE']
+
+  alternate_homes.each do |path|
+    if File.exists?(path)
+      ENV['HOME'] = path
+      found = true
+      break
+    end
+  end
+
+  STDERR.puts <<-EOF
+The HOME (#{old_home}) environment variable was not set to a valid value.
```

Comment:
This seems a little harsh (and not entirely accurate) to throw up in front of people all the time? I think it would be more accurate to say "The HOME environment variable was not set, or was set to an inaccessible location."

However, we should preface this with something that friendly, like "ChefDK needs to establish a consistent HOME directory to store some files, preferably on the local system. The HOME environment variable was not set, or was set to an inaccessible location, so we will attempt to find detect a suitable location."